DSC/dsc.py

```diff
@@ -264,14 +270,27 @@ def install_dsc_packages():
         deb_install_pkg(omi_package_path + '.x64.deb', 'omi', omi_version_deb)
         deb_install_pkg(dsc_package_path + '.x64.deb', 'dsc', dsc_version_deb )
     elif distro_category == DistroCategory.redhat or distro_category == DistroCategory.suse:
-        rpm_install_pkg(omi_package_path + '.x64.rpm', 'omi-' + omi_version_rpm)
-        rpm_install_pkg(dsc_package_path + '.x64.rpm', 'dsc-' + dsc_version_rpm)
+        rpm_install_pkg(omi_package_path + '.x64.rpm', 'omi')
+        rpm_install_pkg(dsc_package_path + '.x64.rpm', 'dsc')
+
+def compare_package_version(system_package_version, package_name):
+    if package_name == 'omi':
+        version = re.match('(\d+).(\d+).(\d+)', system_package_version)
```

Comment:
this regex match for omi and dsc seems duplicate. You should be able to create a separate function that takes in a major/minor version and then invoke that here with omi and dsc values.